externals/kyuubi-chat-engine/src/main/scala/org/apache/kyuubi/engine/chat/provider/ChatGPTProvider.scala

```diff
@@ -75,29 +74,19 @@ class ChatGPTProvider(conf: KyuubiConf) extends ChatProvider {
 
   override def ask(sessionId: String, q: String): String = {
     val messages = chatHistory.get(sessionId)
-    messages.addLast(Message("user", q))
-
-    val request = new HttpPost("https://api.openai.com/v1/chat/completions")
-    val req = Map(
-      "messages" -> messages,
-      "model" -> "gpt-3.5-turbo",
-      "max_tokens" -> 200,
-      "temperature" -> 0.5,
-      "top_p" -> 1)
-    val entity = new StringEntity(mapper.writeValueAsString(req), ContentType.APPLICATION_JSON)
-    request.setEntity(entity)
-    request.setConfig(requestConfig)
-    val response = httpClient.execute(request)
-    val respJson = mapper.readTree(EntityUtils.toString(response.getEntity))
-    response.getStatusLine.getStatusCode match {
-      case HttpStatus.SC_OK =>
-        val replyMessage = mapper.treeToValue[Message](
-          respJson.get("choices").get(0).get("message"))
-        messages.addLast(replyMessage)
-        replyMessage.content
-      case errorStatusCode =>
+    try {
+      messages.addLast(new ChatMessage("user", q))
+      val completionRequest = ChatCompletionRequest.builder()
+        .messages(messages.asScala.toList.asJava)
+        .model("gpt-3.5-turbo")
```

Comment:
let's make it configurable in next PR :)

Reply:
Sure. Make it configurable of other ChatCompletionRequest settings like model name, max tokens, etc.